src/components/modals/TradePreviewModalGP.vue

```diff
@@ -1,70 +1,189 @@
+              </div>
+              <div>
+                <div class="font-medium">
+                  {{ fNum(trading.tokenInAmountInput.value, 'token') }}
+                  {{ trading.tokenIn.value.symbol }}
+                </div>
+                <div class="text-gray-500 text-sm">
+                  {{ tokenInFiatValue }}
+                </div>
+              </div>
+            </div>
+          </div>
+          <div class="arrow-down">
+            <ArrowDownIcon />
+          </div>
+          <div class="p-3">
+            <div class="flex items-center">
+              <div class="mr-3">
+                <BalAsset
+                  :address="trading.tokenOut.value.address"
+                  :size="36"
+                />
+              </div>
+              <div>
+                <div class="font-medium">
+                  {{ fNum(trading.tokenOutAmountInput.value, 'token') }}
+                  {{ trading.tokenOut.value.symbol }}
+                </div>
+                <div class="text-gray-500 text-sm">
+                  {{ tokenOutFiatValue }}
+                  <span v-if="trading.isBalancerTrade.value">
+                    / {{ $t('priceImpact') }}:
+                    {{ fNum(trading.sor.priceImpact.value, 'percent') }}</span
+                  >
+                </div>
+              </div>
+            </div>
           </div>
-          <div class="text-gray-500 text-sm">{{ fNum(valueIn, 'usd') }}</div>
         </div>
-      </div>
-      <div>
-        <div v-if="formattedFeeAmount > 0" class="mt-6 mb-3 text-sm">
-          Fee: {{ fNum(formattedFeeAmount, 'token') }}
-          {{ symbolIn }}
+      </BalCard>
+      <BalCard shadow="none" class="my-5" v-if="showTradeRoute">
+        <TradeRoute
+          :address-in="trading.tokenIn.value.address"
+          :amount-in="trading.tokenInAmountInput.value"
+          :address-out="trading.tokenOut.value.address"
+          :amount-out="trading.tokenOutAmountInput.value"
+          :pools="trading.sor.pools.value"
+          :sor-return="trading.sor.sorReturn.value"
+        />
+      </BalCard>
+      <BalCard noPad shadow="none" class="mb-6" v-if="showSummary">
+        <template v-slot:header>
+          <div class="p-3 flex w-full items-center justify-between border-b">
+            <div class="font-semibold">{{ $t('summary') }}</div>
+            <div class="flex divide-x text-sm">
+              <div
+                :class="!showSummaryInFiat && 'text-blue-600 font-medium'"
+                class="pr-1 cursor-pointer"
+                @click="showSummaryInFiat = false"
+              >
+                {{ $t('tokens') }}
+              </div>
+              <div
+                :class="showSummaryInFiat && 'text-blue-600 font-medium'"
+                class="pl-1 cursor-pointer"
+                @click="showSummaryInFiat = true"
+              >
+                {{ FiatCurrency.USD }}
+              </div>
+            </div>
```

Comment:
These are essential text btns (could potentially be a Bal component) and so it kind of makes sense to uppercase them both and have font-medium in all states, so they don't look like regular text at all. The other reason for keeping font-medium in both states is so you don't get the small shift when you select one or the other.

Finally, I'm really not a fan of having multiple class attributes, so I've cleaned that up.
```suggestion
            <div class="flex divide-x text-xs uppercase">
              <div
                :class="[
                  'pr-1 cursor-pointer font-medium',
                  { 'text-blue-600': !showSummaryInFiat }
                ]"
                @click="showSummaryInFiat = false"
              >
                {{ $t('tokens') }}
              </div>
              <div
                :class="[
                  'pl-1 cursor-pointer font-medium',
                  { 'text-blue-600': showSummaryInFiat }
                ]"
                @click="showSummaryInFiat = true"
              >
                {{ FiatCurrency.USD }}
              </div>
            </div>
```